server/src/main/java/org/opensearch/cluster/routing/remote/InternalRemoteRoutingTableService.java

```diff
@@ -190,6 +210,39 @@ public CheckedRunnable<IOException> getIndexRoutingAsyncAction(
         return () -> uploadIndex(indexRouting, fileName, blobContainer, completionListener);
     }
 
+    public CheckedRunnable<IOException> getIndexRoutingDiffAsyncAction(
+        ClusterState clusterState,
+        Map<String, Diff<IndexRoutingTable>> indexRoutingTableDiff,
+        LatchedActionListener<ClusterMetadataManifest.UploadedMetadata> latchedActionListener,
+        BlobPath clusterBasePath
+    ) {
+
+        BlobPath indexRoutingDiffPath = clusterBasePath.add(INDEX_ROUTING_DIFF_PATH_TOKEN);
+        BlobPath path = pathType.path(
+            RemoteStorePathStrategy.PathInput.builder().basePath(indexRoutingDiffPath).indexUUID(clusterState.getMetadata().clusterUUID()).build(),
```

Comment:
This doesnt look right. Why are we passing clusterUUID as param to indexUUID?

Reply:
As discussed offline, will use base path to IndexShardDiffFile.